contrib/pkg/clusterpool/clusterclaim.go

```diff
@@ -0,0 +1,100 @@
+package clusterpool
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/openshift/hive/contrib/pkg/utils"
+	"github.com/openshift/hive/pkg/apis"
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+type ClusterClaimOptions struct {
+	Name            string
+	Namespace       string
+	Lifetime        time.Duration
+	ClusterPoolName string
+}
+
+func NewClaimClusterPoolCommand() *cobra.Command {
+	opt := &ClusterClaimOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "claim CLUSTER_POOL_NAME CLAIM_NAME",
+		Short: "claims a cluster from a ClusterPool",
+		Long:  "claims a cluster from the ClusterPool in the given namespace",
+		Args:  cobra.ExactArgs(2),
+		Run: func(cmd *cobra.Command, args []string) {
+			log.SetLevel(log.InfoLevel)
+			ctx := context.Background()
+			opt.Name = args[0]
+			err := opt.run(ctx)
+			if err != nil {
+				log.WithContext(ctx).WithError(err).Fatal("Error")
```

Comment:
then drop the WithContext(ctx) here, I don't think we typically use WithContext. We may refer to contextual loggers though but this may have gotten confused with the go context. By contextual logger we just mean a logger that's carrying fields we added and passed around.

Reply:
Thank you this was way easier and made sense for the usecase